### ISSUE TITLE
[Installation] Missing sharp.py

### ISSUE THREAD

**Issue by zatkins2** [simonsobs/pixell#84]
* pixell version: 0.8.0
* Python version: 3.7.6
* Operating System: Ubuntu 18.04 (operating via WSL on Windows 10)

### Description

User installation of pixell (local) in anaconda env. Env running python 3.7.6, automate installed; Ubuntu running gfortran. After completion of setup.py script, sharp.py module not in pixell package. 

I suspect this is related to the very first error message printed after running setup.py, namely:

/path/to/anaconda3/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'library_dirs'
  warnings.warn(msg)

This is related to the `setup(...)` call (which ultimately passes to python `distutils.core.setup(...)` through `setuptools` and `numpy`) not accepting the `'library_dirs'` kwarg. This could be related to the recent numpy 1.18 release (doubtful). More likely it is something to do with the interplay of my local python environment in anaconda, Ubuntu (running via WSL), and some C and/or fortran issue, but I can't seem to fix it after due diligence on stackoverflow.

Of course, the issue (missing sharp.py) may not be related to this warning at all. If this is something you've seen before, or are keen on looking into, then I'm happy to work on it! Otherwise, will be using the Docker image and remote installations of pixell.

### What I Did

Please see link for output.

```
$ python setup.py install --user

https://pastebin.com/54Cet2U8
```

**Comment by msyriac** [comment]
Thanks for reporting. I wasn't able to reproduce any problems with sharp on Linux / Python 3.8.2 / numpy 1.18.3 / gcc 9.3.0. Could you clarify exactly what the problem you face is? sharp.py is not an actual module in the pixell package. There's a cython module that is compiled into a ".so" file, that should end up in your package directory if it is successfully compiled. If you do,
```
cd ~
python
> from pixell import sharp
```
do you get any errors?

**Comment by zatkins2** [comment]
Hi Mat, thanks for getting back so quickly! Based on your feedback, I'm now not sure there is actually an issue ha:

1. If I run `py.test` above my pixell installation, I get:

![image](https://user-images.githubusercontent.com/46325652/80638983-54431580-8a2f-11ea-8f84-48631a2df422.png)

but looking closer, this seems to be some problem with py.test searching only the pixell `__init__.py` file, and not the rest of the package contents (it's an `ImportError` not a `ModuleNotFound` error...doh). More to your point:

2. Running `from pixell import sharp` anywhere in Ubuntu works, as does importing other modules that in turn import sharp...so I think I may have been thrown off by the `py.test` result. Like you mentioned, I do see the `.so` file in the actual `site-packages` for python. 

3. So I think the only issue is: there could be some problem that `py.test` would catch, if it didn't break when running `from pixell import sharp`. But, this doesn't seem like a "big deal?"

Thanks,
Zach

**Comment by msyriac** [comment]
Thanks Zach, indeed getting tests to run correctly when an "in place" build was not done has been non-trivial. My misadventures with this seem to be documented here: https://github.com/simonsobs/pixell/pull/67 -- which suggests "python setup.py test" should work. Does it? If yes, I should update docs to suggest people do that.

**Comment by msyriac** [comment]
("python setup.py test" does work for me. Btw, expect the tests to fail -- there seems to be an unresolved FITS WCS issue, which I've checked is not critical, but is causing the tests to fail at the moment)

**Comment by zatkins2** [comment]
Hi Mat, cool, that did work. I'm actually not sure anything failed; there were some warnings related to FITS files but generally the output was hard to parse -- if you're curious, here it is: 
https://pastebin.com/9ATdDG3n
I'll proceed though. This can probably be closed! Thanks for your help:)

**Comment by msyriac** [comment]
Oh cool, the fact that the tests are not failing for you even though you are using pixell v0.8.0 could help me track down why they are failing for me. So if you don't mind -- could you send me the output of "pip list" and "conda list" in a pastebin?

**Comment by zatkins2** [comment]
Sure. I also found the tests fail in a different environment for me, so I'll paste from both.

**working env**
pip list: https://pastebin.com/eDBttNML
conda list: https://pastebin.com/H3RxepEz

**failing env**
pip list: https://pastebin.com/hEVehsHQ
conda list: https://pastebin.com/RkjJHT7M

The working env is a clone of the base anaconda distribution, with `healpy` installed. The failing env is created via `conda create -n name_of_env anaconda` and then installing `healpy`. The Ubuntu environment is the base distribution with `gfortran`, the `build-essential` pack, and `automake` installed.

**Comment by msyriac** [comment]
Thanks, this helped identify the problem as astropy 4.0 -> 4.0.1.post1 .
I'll close this and investigate further.
